client/src/pages/Explore.tsx

```diff
@@ -13,12 +14,13 @@ export const Explore = () => {
   const [courses, setCourses] = useState<Course[]>([]);
   const [hasMore, setHasMore] = useState(true);
   const [offset, setOffset] = useState(limit);
+  const [error, setError] = useState(false);
 
   useEffect(() => {
     fetchClient
       .getData<Course[]>(`/courses?limit=${limit}`)
       .then((data) => setCourses(data))
-      .catch((err) => console.log(err));
+      .catch((err) => setError(true));
```

Comment:
Does `err` contain anything useful? If so we can pass it to the alert component

Reply:
hmm, i actually haven't tried getting an error from this specific route. I'll get back to you.

Reply:
I think fetch only catches network errors, do you wanna display them?